src/client/clientenvironment.cpp

```diff
@@ -308,7 +308,17 @@ void ClientEnvironment::step(float dtime)
 		node_at_lplayer = m_map->getNode(p);
 
 		u16 light = getInteriorLight(node_at_lplayer, 0, m_client->ndef());
+
+		ItemStack wielditem;
+		wielditem = lplayer->getWieldedItem(&wielditem, nullptr);
+		ItemDefinition wielddef = wielditem.getDefinition(m_client->getItemDefManager());
+
+		u8 wieldlight = decode_light(wielddef.light_source);
+
 		final_color_blend(&lplayer->light_color, light, day_night_ratio);
+
+		if (wielddef.light_source > 0 && wieldlight > lplayer->light_color.getLuminance())
+			lplayer->light_color = irr::video::SColor(255,wieldlight,wieldlight,wieldlight);
```

Comment:
I don't think the idea was to override the light_color from here.
Also this code is below `final_color_blend` which would have applied the changes.

Reply:
From what I can see `final_color_blend` just writes the result to `lplayer->light_color`

Reply:
I'll let you mark as resolved unless there's a problem here

Reply:
bump